File: WasteVisionWebFE/src/data/dto/prediction-dto.ts
import { RoboflowModel } from "@/core/domain/RoboflowModel";
import { User } from "@/core/domain/User";




export interface PredictionDTO {
  id?: string;
  userId: string;
  modelId: string;
  modelName?: string;
  userName?: string;
  originalImageUrl: string;
  processedImageUrl: string;
  date: string;
  user?: User;
  roboflowModel?: RoboflowModel 
}

export interface PredictionSearchParamsDTO {
  userId?: string;
  modelName?: string;
  userName?: string;
  dateFrom?: string;
  dateTo?: string;
}

File: WasteVisionWebBE/Domain/Prediction/PredictionDto.cs
using System;

namespace DDDSample1.Domain.Predictions
{
    public class PredictionDto
    {
        public Guid Id { get; set; }
        public string UserId { get; set; }
        public string ModelId { get; set; }
        public string OriginalImageUrl { get; set; }
        public string ProcessedImageUrl { get; set; }
        public string Date { get; set; }

        public PredictionDto(Guid id, string userId, string modelId,
            string originalImageUrl, string processedImageUrl,
            string date)
        {
            Id = id;
            UserId = userId;
            ModelId = modelId;
            OriginalImageUrl = originalImageUrl;
            ProcessedImageUrl = processedImageUrl;
            Date = date;

        }
    }
}